src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -205,17 +208,34 @@ public override long Seek(long offset, SeekOrigin origin)
 
         public override void SetLength(long value)
         {
-            if (InnerStream.Length < value)
-                InnerStream.SetLength(value);
+            try
+            {
+                if (InnerStream.Length < value)
+                    InnerStream.SetLength(value);
 
-            _length = value;
+                _length = value;
+            }
+            catch (IOException e)
+            {
+                if (e.IsOutOfDiskSpaceException())
+                {
+                    IndexingUtils.ThrowDiskFullException(InnerStream.Name);
+                }
+            }
```

Comment:
you are not rethrowing an exception when this is not out of disk space one, is this intended?

Reply:
Done